### ISSUE TITLE
SegFault & hang on macOS 14.4

### ISSUE THREAD

**Issue by SichangHe** [ahrm/sioyek#1262]
Just tried the Mac ARM build from [Release fa863d](https://github.com/hexomancer/sioyek/releases/tag/manual_release19) on macOS 14. Cannot launch it due to SegFault on dereferencing a null pointer. I will attach the crash report below, if it helps.

I also tried the Mac non-ARM build. It also SegFaults the same.

This is after #1169.

**Comment by SichangHe** [comment]
<details><summary>Crash report from Apple.</summary>

```sh
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               sioyek [52241]
Path:                  /Applications/sioyek.app/Contents/MacOS/sioyek
Identifier:            info.sioyek.sioyek
Version:               2.0 (???)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2024-12-19 20:31:58.8696 -0800
OS Version:            macOS 14.4 (23E214)
Report Version:        12
Anonymous UUID:        3159813C-418C-EF15-55C0-36C47CD7F72E

Sleep/Wake UUID:       A4F35F1D-78F7-4E16-927B-FF902A2EF860

Time Awake Since Boot: 56000 seconds
Time Since Wake:       732 seconds

System Integrity Protection: disabled

Crashed Thread:        11

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Codes:       0x0000000000000001, 0x0000000000000000

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [52241]

VM Region Info: 0 is not in any region.  Bytes before following region: 4340383744
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      102b50000-10548c000    [ 41.2M] r-x/r-x SM=COW  /Applications/sioyek.app/Contents/MacOS/sioyek

Thread 0::  Dispatch queue: com.apple.main-thread
0   libsystem_platform.dylib      	       0x1890a7354 _platform_memmove + 436
1   sioyek                        	       0x102f66ba0 pdf_new_string + 80
2   sioyek                        	       0x102f89080 pdf_parse_ind_obj_or_newobj + 448
3   sioyek                        	       0x102f9f140 pdf_cache_object + 696
4   sioyek                        	       0x102f9ebbc pdf_resolve_indirect + 128
5   sioyek                        	       0x102f9fd5c pdf_resolve_indirect_chain + 44
6   sioyek                        	       0x102f673f0 pdf_to_text_string + 52
7   sioyek                        	       0x102f82fbc 0x102b50000 + 4403132
8   sioyek                        	       0x102eed4f4 0x102b50000 + 3790068
9   sioyek                        	       0x102eed5fc 0x102b50000 + 3790332
10  sioyek                        	       0x102eed424 fz_load_outline_from_iterator + 88
11  sioyek                        	       0x102ba746c Document::get_toc_outline() + 68
12  sioyek                        	       0x102ba74c8 Document::create_toc_tree(std::__1::vector<TocNode*, std::__1::allocator<TocNode*>>&) + 64
13  sioyek                        	       0x102ba9210 Document::open(bool*, bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool) + 340
14  sioyek                        	       0x102be2cb0 DocumentView::open_document(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t>> const&, bool*, bool, std::__1::optional<OpenedBookState>, bool) + 116
15  sioyek                        	       0x102cc012c MainWidget::open_document(Path const&, std::__1::optional<float>, std::__1::optional<float>, std::__1::optional<float>) + 176
16  sioyek                        	       0x102cab784 handle_args(QList<QString> const&, QLocalSocket*) + 6980
17  sioyek                        	       0x102cad238 main + 4640
18  dyld                          	       0x188cee0e0 start + 2360

Thread 1:
0   libsystem_pthread.dylib       	       0x189071d20 start_wqthread + 0

Thread 2:
0   libsystem_pthread.dylib       	       0x189071d20 start_wqthread + 0

Thread 3:
0   libsystem_pthread.dylib       	       0x189071d20 start_wqthread + 0

Thread 4:: com.apple.CFSocket.private
0   libsystem_kernel.dylib        	       0x189041358 __select + 8
1   CoreFoundation                	       0x18917f30c __CFSocketManager + 640
2   libsystem_pthread.dylib       	       0x189076f94 _pthread_start + 136
3   libsystem_pthread.dylib       	       0x189071d34 thread_start + 8

Thread 5:
0   libsystem_kernel.dylib        	       0x1890397e8 __semwait_signal + 8
1   libsystem_c.dylib             	       0x188f1a274 nanosleep + 220
2   sioyek                        	       0x102e2be40 sleep_ms(unsigned int) + 64
3   sioyek                        	       0x102d385d4 PdfRenderer::run(int) + 524
4   sioyek                        	       0x102d39b4c 0x102b50000 + 2005836
5   libsystem_pthread.dylib       	       0x189076f94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x189071d34 thread_start + 8

Thread 6:
0   libsystem_kernel.dylib        	       0x1890397e8 __semwait_signal + 8
1   libsystem_c.dylib             	       0x188f1a274 nanosleep + 220
2   sioyek                        	       0x102e2be40 sleep_ms(unsigned int) + 64
3   sioyek                        	       0x102d385d4 PdfRenderer::run(int) + 524
4   sioyek                        	       0x102d39b4c 0x102b50000 + 2005836
5   libsystem_pthread.dylib       	       0x189076f94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x189071d34 thread_start + 8

Thread 7:
0   libsystem_kernel.dylib        	       0x1890397e8 __semwait_signal + 8
1   libsystem_c.dylib             	       0x188f1a274 nanosleep + 220
2   sioyek                        	       0x102e2be40 sleep_ms(unsigned int) + 64
3   sioyek                        	       0x102d385d4 PdfRenderer::run(int) + 524
4   sioyek                        	       0x102d39b4c 0x102b50000 + 2005836
5   libsystem_pthread.dylib       	       0x189076f94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x189071d34 thread_start + 8

Thread 8:
0   libsystem_kernel.dylib        	       0x1890397e8 __semwait_signal + 8
1   libsystem_c.dylib             	       0x188f1a274 nanosleep + 220
2   sioyek                        	       0x102e2be40 sleep_ms(unsigned int) + 64
3   sioyek                        	       0x102d385d4 PdfRenderer::run(int) + 524
4   sioyek                        	       0x102d39b4c 0x102b50000 + 2005836
5   libsystem_pthread.dylib       	       0x189076f94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x189071d34 thread_start + 8

Thread 9:
0   libsystem_kernel.dylib        	       0x1890397e8 __semwait_signal + 8
1   libsystem_c.dylib             	       0x188f1a274 nanosleep + 220
2   sioyek                        	       0x102e2be40 sleep_ms(unsigned int) + 64
3   sioyek                        	       0x102d372e8 PdfRenderer::run_search(int) + 148
4   sioyek                        	       0x102d39c24 0x102b50000 + 2006052
5   libsystem_pthread.dylib       	       0x189076f94 _pthread_start + 136
6   libsystem_pthread.dylib       	       0x189071d34 thread_start + 8

Thread 10:
0   libsystem_pthread.dylib       	       0x189071d20 start_wqthread + 0

Thread 11 Crashed:
0   libsystem_platform.dylib      	       0x1890a4904 _platform_strlen + 4
1   sioyek                        	       0x102b8a8b0 0x102b50000 + 239792
2   sioyek                        	       0x102d65874 sqlite3_exec + 404
3   sioyek                        	       0x102b8a660 DatabaseManager::select_mark(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::vector<Mark, std::__1::allocator<Mark>>&) + 260
4   sioyek                        	       0x102ba1b64 Document::load_document_metadata_from_db() + 504
5   sioyek                        	       0x102baa144 0x102b50000 + 368964
6   sioyek                        	       0x102bcd388 0x102b50000 + 512904
7   libsystem_pthread.dylib       	       0x189076f94 _pthread_start + 136
8   libsystem_pthread.dylib       	       0x189071d34 thread_start + 8


Thread 11 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000008f07   x3: 0x0000000000008f07
    x4: 0x0000000000000000   x5: 0x0000000000000000   x6: 0x0000000000000030   x7: 0x0000000000000000
    x8: 0x1a65a0c5f23f00e8   x9: 0x1a65a0c5f23f00e8  x10: 0x0000000188f85910  x11: 0x00000000ffffffd0
   x12: 0x0000000000010001  x13: 0x0000000000000000  x14: 0x0000000000192e77  x15: 0x0000000000000000
   x16: 0x00000001890a4900  x17: 0x00000001fb0ef788  x18: 0x0000000000000000  x19: 0x00000001578a2460
   x20: 0x000000012815f5e8  x21: 0x000000012815f5b0  x22: 0x0000000000000000  x23: 0x00000001578a2460
   x24: 0x0000000102b8a810  x25: 0x0000000000000000  x26: 0x000000000000002f  x27: 0x0000000000000000
   x28: 0x7fffffffffffffef   fp: 0x000000016d8aeb40   lr: 0x0000000102b8a8b0
    sp: 0x000000016d8aea10   pc: 0x00000001890a4904 cpsr: 0x60001000
   far: 0x0000000000000000  esr: 0x92000006 (Data Abort) byte read Translation fault

Binary Images:
       0x11387c000 -        0x113883fff libqmacjp2.dylib (*) <bab7ee3d-ce5e-37d6-8d6e-f552aea36391> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqmacjp2.dylib
       0x107448000 -        0x10744ffff libqtga.dylib (*) <f42378cf-c9c9-36b0-9c4c-5866dc40ff86> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqtga.dylib
       0x107434000 -        0x10743bfff libqicns.dylib (*) <06500557-4537-3752-875b-707c9ad93bec> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqicns.dylib
       0x108ddc000 -        0x108de3fff libqpdf.dylib (*) <412c12a2-2787-3a4f-a5bd-3f787b904dbe> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqpdf.dylib
       0x116700000 -        0x116ecbfff org.qt-project.QtPdf (6.7) <d0be1ce6-5efc-37a8-90fb-ea9a96f28c44> /Applications/sioyek.app/Contents/Frameworks/QtPdf.framework/Versions/A/QtPdf
       0x107b0c000 -        0x107b13fff libqsvg.dylib (*) <b3999c49-94d9-387d-8813-653d62c9bf4b> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqsvg.dylib
       0x108d70000 -        0x108dcffff libqtiff.dylib (*) <237646a6-00b4-3be3-9fc3-9844626f6fdc> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqtiff.dylib
       0x107a94000 -        0x107afbfff libqjpeg.dylib (*) <f3415f27-b091-3b92-8816-8ae9987fecec> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqjpeg.dylib
       0x10749c000 -        0x1074a3fff libqmacheif.dylib (*) <554c616b-74bb-3aad-9311-4bfdbb6152b9> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqmacheif.dylib
       0x107488000 -        0x10748ffff libqico.dylib (*) <d6b39361-793f-3a6b-bb70-7cdb7bc6c2dc> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqico.dylib
       0x107a18000 -        0x107a83fff libqwebp.dylib (*) <afb09f29-c1c6-3f4a-b548-4ff410a78bed> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqwebp.dylib
       0x107474000 -        0x10747bfff libqwbmp.dylib (*) <0bb6602f-bb41-3a7a-a321-0a58c563a796> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqwbmp.dylib
       0x107460000 -        0x107467fff libqgif.dylib (*) <7016d1af-7056-3787-8937-1af22fe1c883> /Applications/sioyek.app/Contents/PlugIns/imageformats/libqgif.dylib
       0x10740c000 -        0x107417fff libobjc-trampolines.dylib (*) <e8a1b184-0349-3c61-a119-6543eb038317> /usr/lib/libobjc-trampolines.dylib
       0x1073f4000 -        0x1073fffff libqsvgicon.dylib (*) <de7edee1-ba24-3a3d-8b1d-492180cee2ad> /Applications/sioyek.app/Contents/PlugIns/iconengines/libqsvgicon.dylib
       0x106e1c000 -        0x106e3ffff libqmacstyle.dylib (*) <8f9cbcc6-87f7-3528-a266-a333677509fe> /Applications/sioyek.app/Contents/PlugIns/styles/libqmacstyle.dylib
       0x106d80000 -        0x106debfff com.apple.AppleMetalOpenGLRenderer (1.0) <83d94df2-aca8-3695-a3a1-bf8ca16657ac> /System/Library/Extensions/AppleMetalOpenGLRenderer.bundle/Contents/MacOS/AppleMetalOpenGLRenderer
       0x105918000 -        0x1059b7fff libqcocoa.dylib (*) <24e1e52e-027b-3f8f-b02a-288279fcd6d8> /Applications/sioyek.app/Contents/PlugIns/platforms/libqcocoa.dylib
       0x1064c0000 -        0x1064dffff com.apple.security.csparser (3.0) <d5ba6df5-4b0f-3b49-b7e3-923347e0a7d1> /System/Library/Frameworks/Security.framework/Versions/A/PlugIns/csparser.bundle/Contents/MacOS/csparser
       0x105aec000 -        0x105afffff org.qt-project.QtQuickWidgets (6.7) <8775d2c0-7eaf-364a-96eb-7a42e61261d9> /Applications/sioyek.app/Contents/Frameworks/QtQuickWidgets.framework/Versions/A/QtQuickWidgets
       0x106620000 -        0x1069fffff org.qt-project.QtQuick (6.7) <50651b87-23e4-348b-93e1-a9011aa5a221> /Applications/sioyek.app/Contents/Frameworks/QtQuick.framework/Versions/A/QtQuick
       0x105acc000 -        0x105ad7fff org.qt-project.QtOpenGLWidgets (6.7) <17df069b-7d3c-3473-ba95-76fcb0533ca6> /Applications/sioyek.app/Contents/Frameworks/QtOpenGLWidgets.framework/Versions/A/QtOpenGLWidgets
       0x105c44000 -        0x105ca7fff org.qt-project.QtOpenGL (6.7) <71f60eb4-5bda-34ce-ae28-6150eed431fa> /Applications/sioyek.app/Contents/Frameworks/QtOpenGL.framework/Versions/A/QtOpenGL
       0x105cd8000 -        0x10613ffff org.qt-project.QtWidgets (6.7) <45cafaab-5d55-3c78-8832-7f3ef039a017> /Applications/sioyek.app/Contents/Frameworks/QtWidgets.framework/Versions/A/QtWidgets
       0x1062a4000 -        0x1062f3fff org.qt-project.Qt3DInput (6.7) <235765b7-3f1b-3e91-bf20-9a55ef17dbcc> /Applications/sioyek.app/Contents/Frameworks/Qt3DInput.framework/Versions/A/Qt3DInput
       0x106314000 -        0x106363fff org.qt-project.Qt3DCore (6.7) <632041ad-c8d6-36f7-af3b-24212f3ea420> /Applications/sioyek.app/Contents/Frameworks/Qt3DCore.framework/Versions/A/Qt3DCore
       0x106398000 -        0x1063dffff org.qt-project.QtSvg (6.7) <8ab904ac-9e21-34f4-91a2-29c710d87496> /Applications/sioyek.app/Contents/Frameworks/QtSvg.framework/Versions/A/QtSvg
       0x105a90000 -        0x105aa3fff org.qt-project.QtTextToSpeech (6.7) <4263f052-f0d3-3ac7-8448-c644443273a2> /Applications/sioyek.app/Contents/Frameworks/QtTextToSpeech.framework/Versions/A/QtTextToSpeech
       0x105b18000 -        0x105babfff org.qt-project.QtMultimedia (6.7) <4eaae195-9489-39c6-92ce-b578cf11fc7d> /Applications/sioyek.app/Contents/Frameworks/QtMultimedia.framework/Versions/A/QtMultimedia
       0x107b28000 -        0x10817bfff org.qt-project.QtGui (6.7) <6319e5b4-cc0b-3b41-a5fe-d6de1f323bc9> /Applications/sioyek.app/Contents/Frameworks/QtGui.framework/Versions/A/QtGui
       0x10656c000 -        0x1065effff org.qt-project.QtQmlModels (6.7) <6e906706-64f8-3786-bc9e-e6520d871fd6> /Applications/sioyek.app/Contents/Frameworks/QtQmlModels.framework/Versions/A/QtQmlModels
       0x1074bc000 -        0x107857fff org.qt-project.QtQml (6.7) <b8e1bbf6-cbe7-3687-9149-8e1bb9f9039d> /Applications/sioyek.app/Contents/Frameworks/QtQml.framework/Versions/A/QtQml
       0x106e98000 -        0x106fbbfff org.qt-project.QtNetwork (6.7) <441c48bf-0071-3eb6-b856-5e7ba17fb8e1> /Applications/sioyek.app/Contents/Frameworks/QtNetwork.framework/Versions/A/QtNetwork
       0x105ab4000 -        0x105ab7fff org.qt-project.QtConcurrent (6.7) <307c0fe6-1274-34a5-81ea-76221152515c> /Applications/sioyek.app/Contents/Frameworks/QtConcurrent.framework/Versions/A/QtConcurrent
       0x108e24000 -        0x1092cffff org.qt-project.QtCore (6.7) <36d433b3-6671-3d70-b539-a559950fbb64> /Applications/sioyek.app/Contents/Frameworks/QtCore.framework/Versions/A/QtCore
       0x106b78000 -        0x106bf7fff org.qt-project.QtDBus (6.7) <68bb128c-d7f2-3fbe-b35e-b3ea9efacf90> /Applications/sioyek.app/Contents/Frameworks/QtDBus.framework/Versions/A/QtDBus
       0x102b50000 -        0x10548bfff info.sioyek.sioyek (2.0) <7068a90b-3317-3d66-b28e-1c069ed0b77e> /Applications/sioyek.app/Contents/MacOS/sioyek
       0x1890a3000 -        0x1890aafe7 libsystem_platform.dylib (*) <031f3e8c-5227-3138-a444-68174c1c28cf> /usr/lib/system/libsystem_platform.dylib
       0x188ce8000 -        0x188d709db dyld (*) <8e1e5ee2-f89a-33a7-bb0a-74bdc06b7828> /usr/lib/dyld
               0x0 - 0xffffffffffffffff ??? (*) <00000000-0000-0000-0000-000000000000> ???
       0x189070000 -        0x18907cfff libsystem_pthread.dylib (*) <45239f06-cc53-36d0-9933-7776ac7ea2fa> /usr/lib/system/libsystem_pthread.dylib
       0x189035000 -        0x18906fffb libsystem_kernel.dylib (*) <1889ce0a-52e7-3122-8907-81af920ac472> /usr/lib/system/libsystem_kernel.dylib
       0x1890d8000 -        0x1895b0fff com.apple.CoreFoundation (6.9) <33908a83-098f-3437-973e-fb848c4f39df> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
       0x188f0d000 -        0x188f8bff3 libsystem_c.dylib (*) <49477e07-e77b-332f-b98d-79ca210a866d> /usr/lib/system/libsystem_c.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 2
    thread_create: 4
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=931.0M resident=0K(0%) swapped_out_or_unallocated=931.0M(100%)
Writable regions: Total=1.7G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.7G(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Activity Tracing                   256K        1 
ColorSync                          608K       28 
CoreAnimation                       16K        1 
CoreGraphics                        16K        1 
Foundation                          16K        1 
Kernel Alloc Once                   32K        1 
MALLOC                             1.7G       42 
MALLOC guard page                  192K       12 
OpenGL GLSL                        256K        3 
STACK GUARD                       56.2M       12 
Stack                             13.8M       12 
VM_ALLOCATE                        224K       14 
__AUTH                             974K      215 
__AUTH_CONST                      16.1M      382 
__CTF                               824        1 
__DATA                            5086K      407 
__DATA_CONST                      20.4M      420 
__DATA_DIRTY                       931K      117 
__FONT_DATA                        2352        1 
__GLSLBUILTINS                    5174K        1 
__LINKEDIT                       533.4M       38 
__OBJC_RO                         71.7M        2 
__OBJC_RW                         2195K        1 
__TEXT                           397.6M      434 
dyld private memory                272K        2 
mapped file                      182.6M       25 
shared memory                     1360K       14 
===========                     =======  ======= 
TOTAL                              3.0G     2188 



-----------
Full Report
-----------

{"app_name":"sioyek","timestamp":"2024-12-19 20:31:59.00 -0800","app_version":"2.0","slice_uuid":"7068a90b-3317-3d66-b28e-1c069ed0b77e","build_version":"","platform":1,"bundleID":"info.sioyek.sioyek","share_with_app_devs":0,"is_first_party":0,"bug_type":"309","os_version":"macOS 14.4 (23E214)","roots_installed":0,"name":"sioyek","incident_id":"983A50F8-57B4-4771-9599-7A2A4E8552BD"}
{
  "uptime" : 56000,
  "procRole" : "Foreground",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "Mac15,6",
  "coalitionID" : 9298,
  "osVersion" : {
    "train" : "macOS 14.4",
    "build" : "23E214",
    "releaseType" : "User"
  },
  "captureTime" : "2024-12-19 20:31:58.8696 -0800",
  "codeSigningMonitor" : 1,
  "incident" : "983A50F8-57B4-4771-9599-7A2A4E8552BD",
  "pid" : 52241,
  "translated" : false,
  "cpuType" : "ARM-64",
  "roots_installed" : 0,
  "bug_type" : "309",
  "procLaunch" : "2024-12-19 20:31:58.2515 -0800",
  "procStartAbsTime" : 1347424680607,
  "procExitAbsTime" : 1347439493371,
  "procName" : "sioyek",
  "procPath" : "\/Applications\/sioyek.app\/Contents\/MacOS\/sioyek",
  "bundleInfo" : {"CFBundleShortVersionString":"2.0","CFBundleIdentifier":"info.sioyek.sioyek"},
  "storeInfo" : {"deviceIdentifierForVendor":"BCF24D3F-DD72-50A8-AC8E-74CA3493C2FD","thirdParty":true},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "info.sioyek.sioyek",
  "crashReporterKey" : "3159813C-418C-EF15-55C0-36C47CD7F72E",
  "lowPowerMode" : 1,
  "codeSigningID" : "sioyek",
  "codeSigningTeamID" : "",
  "codeSigningFlags" : 570556417,
  "codeSigningValidationCategory" : 10,
  "codeSigningTrustLevel" : 4294967295,
  "instructionByteStream" : {"beforePC":"AagxbiMAJh5AAADLPwAD6yEwg5oAAAGLwANf1h8gA9UfIAPVAex8kg==","atPC":"IADAPcP5\/xBiBME8AgxAkmMAAsthAMA9AByhTgUAABQfIAPVHyAD1Q=="},
  "wakeTime" : 732,
  "sleepWakeUUID" : "A4F35F1D-78F7-4E16-927B-FF902A2EF860",
  "sip" : "disabled",
  "vmRegionInfo" : "0 is not in any region.  Bytes before following region: 4340383744\n      REGION TYPE                    START - END         [ VSIZE] PRT\/MAX SHRMOD  REGION DETAIL\n      UNUSED SPACE AT START\n--->  \n      __TEXT                      102b50000-10548c000    [ 41.2M] r-x\/r-x SM=COW  \/Applications\/sioyek.app\/Contents\/MacOS\/sioyek",
  "exception" : {"codes":"0x0000000000000001, 0x0000000000000000","rawCodes":[1,0],"type":"EXC_BAD_ACCESS","signal":"SIGSEGV","subtype":"KERN_INVALID_ADDRESS at 0x0000000000000000"},
  "termination" : {"flags":0,"code":11,"namespace":"SIGNAL","indicator":"Segmentation fault: 11","byProc":"exc handler","byPid":52241},
  "vmregioninfo" : "0 is not in any region.  Bytes before following region: 4340383744\n      REGION TYPE                    START - END         [ VSIZE] PRT\/MAX SHRMOD  REGION DETAIL\n      UNUSED SPACE AT START\n--->  \n      __TEXT                      102b50000-10548c000    [ 41.2M] r-x\/r-x SM=COW  \/Applications\/sioyek.app\/Contents\/MacOS\/sioyek",
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":4,"thread_set_state":0,"task_for_pid":2},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 11,
  "threads" : [{"id":1650237,"threadState":{"x":[{"value":105553124498904},{"value":5369365032},{"value":18446744073709551614},{"value":105553124498928},{"value":6},{"value":7616},{"value":6989697674165380352},{"value":2496},{"value":7536641},{"value":10995116280322},{"value":10995116280320},{"value":10995116280322},{"value":105553133450780},{"value":2095104},{"value":2043},{"value":2472558712},{"value":6594130336,"symbolLocation":0,"symbol":"_platform_memmove"},{"value":8507014744},{"value":0},{"value":30},{"value":5369365008},{"value":105553124498880},{"value":105553124498904},{"value":105553168012048},{"value":5369364960},{"value":5369364480},{"value":6126496404},{"value":0},{"value":53}],"flavor":"ARM_THREAD_STATE64","lr":{"value":4344671136},"cpsr":{"value":2147487744},"fp":{"value":6126496208},"sp":{"value":6126496176},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6594130772},"far":{"value":0}},"queue":"com.apple.main-thread","frames":[{"imageOffset":17236,"symbol":"_platform_memmove","symbolLocation":436,"imageIndex":37},{"imageOffset":4287392,"symbol":"pdf_new_string","symbolLocation":80,"imageIndex":36},{"imageOffset":4427904,"symbol":"pdf_parse_ind_obj_or_newobj","symbolLocation":448,"imageIndex":36},{"imageOffset":4518208,"symbol":"pdf_cache_object","symbolLocation":696,"imageIndex":36},{"imageOffset":4516796,"symbol":"pdf_resolve_indirect","symbolLocation":128,"imageIndex":36},{"imageOffset":4521308,"symbol":"pdf_resolve_indirect_chain","symbolLocation":44,"imageIndex":36},{"imageOffset":4289520,"symbol":"pdf_to_text_string","symbolLocation":52,"imageIndex":36},{"imageOffset":4403132,"imageIndex":36},{"imageOffset":3790068,"imageIndex":36},{"imageOffset":3790332,"imageIndex":36},{"imageOffset":3789860,"symbol":"fz_load_outline_from_iterator","symbolLocation":88,"imageIndex":36},{"imageOffset":357484,"symbol":"Document::get_toc_outline()","symbolLocation":68,"imageIndex":36},{"imageOffset":357576,"symbol":"Document::create_toc_tree(std::__1::vector<TocNode*, std::__1::allocator<TocNode*>>&)","symbolLocation":64,"imageIndex":36},{"imageOffset":365072,"symbol":"Document::open(bool*, bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool)","symbolLocation":340,"imageIndex":36},{"imageOffset":601264,"symbol":"DocumentView::open_document(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t>> const&, bool*, bool, std::__1::optional<OpenedBookState>, bool)","symbolLocation":116,"imageIndex":36},{"imageOffset":1507628,"symbol":"MainWidget::open_document(Path const&, std::__1::optional<float>, std::__1::optional<float>, std::__1::optional<float>)","symbolLocation":176,"imageIndex":36},{"imageOffset":1423236,"symbol":"handle_args(QList<QString> const&, QLocalSocket*)","symbolLocation":6980,"imageIndex":36},{"imageOffset":1430072,"symbol":"main","symbolLocation":4640,"imageIndex":36},{"imageOffset":24800,"symbol":"start","symbolLocation":2360,"imageIndex":38}]},{"id":1650256,"frames":[{"imageOffset":7456,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":40}],"threadState":{"x":[{"value":6127628288},{"value":6923},{"value":6127091712},{"value":0},{"value":409602},{"value":18446744073709551615},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0}],"flavor":"ARM_THREAD_STATE64","lr":{"value":0},"cpsr":{"value":4096},"fp":{"value":0},"sp":{"value":6127628288},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593912096},"far":{"value":0}}},{"id":1650257,"frames":[{"imageOffset":7456,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":40}],"threadState":{"x":[{"value":6128201728},{"value":4867},{"value":6127665152},{"value":0},{"value":409604},{"value":18446744073709551615},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0}],"flavor":"ARM_THREAD_STATE64","lr":{"value":0},"cpsr":{"value":4096},"fp":{"value":0},"sp":{"value":6128201728},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593912096},"far":{"value":0}}},{"id":1650267,"frames":[{"imageOffset":7456,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":40}],"threadState":{"x":[{"value":6128775168},{"value":19203},{"value":6128238592},{"value":0},{"value":409604},{"value":18446744073709551615},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0}],"flavor":"ARM_THREAD_STATE64","lr":{"value":0},"cpsr":{"value":4096},"fp":{"value":0},"sp":{"value":6128775168},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593912096},"far":{"value":0}}},{"id":1650282,"name":"com.apple.CFSocket.private","threadState":{"x":[{"value":4},{"value":0},{"value":105553133482144},{"value":0},{"value":0},{"value":96},{"value":0},{"value":0},{"value":6127055072},{"value":140737488355320},{"value":52118},{"value":48},{"value":105553133442416},{"value":2095104},{"value":2043},{"value":2450092035},{"value":93},{"value":8507016104},{"value":0},{"value":32},{"value":8335778664,"symbolLocation":0,"symbol":"__CFActiveSocketsLock"},{"value":1},{"value":0},{"value":0},{"value":105553133482144},{"value":105553133482560},{"value":8417217536,"symbolLocation":0,"symbol":"__kCFNull"},{"value":0},{"value":8335777792,"symbolLocation":16,"symbol":"OBJC_CLASS_$_NSConstantArray"}],"flavor":"ARM_THREAD_STATE64","lr":{"value":6595015436},"cpsr":{"value":1610616832},"fp":{"value":6127054784},"sp":{"value":6127021008},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593712984},"far":{"value":0}},"frames":[{"imageOffset":50008,"symbol":"__select","symbolLocation":8,"imageIndex":41},{"imageOffset":684812,"symbol":"__CFSocketManager","symbolLocation":640,"imageIndex":42},{"imageOffset":28564,"symbol":"_pthread_start","symbolLocation":136,"imageIndex":40},{"imageOffset":7476,"symbol":"thread_start","symbolLocation":8,"imageIndex":40}]},{"id":1650289,"frames":[{"imageOffset":18408,"symbol":"__semwait_signal","symbolLocation":8,"imageIndex":41},{"imageOffset":53876,"symbol":"nanosleep","symbolLocation":220,"imageIndex":43},{"imageOffset":2997824,"symbol":"sleep_ms(unsigned int)","symbolLocation":64,"imageIndex":36},{"imageOffset":2000340,"symbol":"PdfRenderer::run(int)","symbolLocation":524,"imageIndex":36},{"imageOffset":2005836,"imageIndex":36},{"imageOffset":28564,"symbol":"_pthread_start","symbolLocation":136,"imageIndex":40},{"imageOffset":7476,"symbol":"thread_start","symbolLocation":8,"imageIndex":40}],"threadState":{"x":[{"value":4},{"value":0},{"value":1},{"value":1},{"value":0},{"value":100000000},{"value":0},{"value":0},{"value":8335707576,"symbolLocation":0,"symbol":"clock_sem"},{"value":3},{"value":17},{"value":2},{"value":0},{"value":0},{"value":2},{"value":4294967295},{"value":334},{"value":8507026528},{"value":0},{"value":0},{"value":6129347792},{"value":4697653248},{"value":0},{"value":0},{"value":0},{"value":4949297864},{"value":0},{"value":6129348240},{"value":4949297992}],"flavor":"ARM_THREAD_STATE64","lr":{"value":6592504436},"cpsr":{"value":2684358656},"fp":{"value":6129347776},"sp":{"value":6129347728},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593681384},"far":{"value":0}}},{"id":1650290,"frames":[{"imageOffset":18408,"symbol":"__semwait_signal","symbolLocation":8,"imageIndex":41},{"imageOffset":53876,"symbol":"nanosleep","symbolLocation":220,"imageIndex":43},{"imageOffset":2997824,"symbol":"sleep_ms(unsigned int)","symbolLocation":64,"imageIndex":36},{"imageOffset":2000340,"symbol":"PdfRenderer::run(int)","symbolLocation":524,"imageIndex":36},{"imageOffset":2005836,"imageIndex":36},{"imageOffset":28564,"symbol":"_pthread_start","symbolLocation":136,"imageIndex":40},{"imageOffset":7476,"symbol":"thread_start","symbolLocation":8,"imageIndex":40}],"threadState":{"x":[{"value":4},{"value":0},{"value":1},{"value":1},{"value":0},{"value":100000000},{"value":0},{"value":0},{"value":8335707576,"symbolLocation":0,"symbol":"clock_sem"},{"value":3},{"value":17},{"value":2},{"value":0},{"value":0},{"value":2},{"value":4294967295},{"value":334},{"value":8507026528},{"value":0},{"value":0},{"value":6129921232},{"value":5905612800},{"value":0},{"value":0},{"value":0},{"value":4949297864},{"value":0},{"value":6129921680},{"value":4949297992}],"flavor":"ARM_THREAD_STATE64","lr":{"value":6592504436},"cpsr":{"value":2684358656},"fp":{"value":6129921216},"sp":{"value":6129921168},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593681384},"far":{"value":0}}},{"id":1650291,"frames":[{"imageOffset":18408,"symbol":"__semwait_signal","symbolLocation":8,"imageIndex":41},{"imageOffset":53876,"symbol":"nanosleep","symbolLocation":220,"imageIndex":43},{"imageOffset":2997824,"symbol":"sleep_ms(unsigned int)","symbolLocation":64,"imageIndex":36},{"imageOffset":2000340,"symbol":"PdfRenderer::run(int)","symbolLocation":524,"imageIndex":36},{"imageOffset":2005836,"imageIndex":36},{"imageOffset":28564,"symbol":"_pthread_start","symbolLocation":136,"imageIndex":40},{"imageOffset":7476,"symbol":"thread_start","symbolLocation":8,"imageIndex":40}],"threadState":{"x":[{"value":4},{"value":0},{"value":1},{"value":1},{"value":0},{"value":100000000},{"value":0},{"value":0},{"value":8335707576,"symbolLocation":0,"symbol":"clock_sem"},{"value":3},{"value":17},{"value":2},{"value":0},{"value":0},{"value":2},{"value":4294967295},{"value":334},{"value":8507026528},{"value":0},{"value":0},{"value":6130494672},{"value":6308265984},{"value":0},{"value":0},{"value":0},{"value":4949297864},{"value":0},{"value":6130495120},{"value":4949297992}],"flavor":"ARM_THREAD_STATE64","lr":{"value":6592504436},"cpsr":{"value":2684358656},"fp":{"value":6130494656},"sp":{"value":6130494608},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593681384},"far":{"value":0}}},{"id":1650292,"frames":[{"imageOffset":18408,"symbol":"__semwait_signal","symbolLocation":8,"imageIndex":41},{"imageOffset":53876,"symbol":"nanosleep","symbolLocation":220,"imageIndex":43},{"imageOffset":2997824,"symbol":"sleep_ms(unsigned int)","symbolLocation":64,"imageIndex":36},{"imageOffset":2000340,"symbol":"PdfRenderer::run(int)","symbolLocation":524,"imageIndex":36},{"imageOffset":2005836,"imageIndex":36},{"imageOffset":28564,"symbol":"_pthread_start","symbolLocation":136,"imageIndex":40},{"imageOffset":7476,"symbol":"thread_start","symbolLocation":8,"imageIndex":40}],"threadState":{"x":[{"value":4},{"value":0},{"value":1},{"value":1},{"value":0},{"value":100000000},{"value":0},{"value":0},{"value":8335707576,"symbolLocation":0,"symbol":"clock_sem"},{"value":3},{"value":17},{"value":2},{"value":0},{"value":0},{"value":2},{"value":4294967295},{"value":334},{"value":8507026528},{"value":0},{"value":0},{"value":6131068112},{"value":6174048256},{"value":0},{"value":0},{"value":0},{"value":4949297864},{"value":0},{"value":6131068560},{"value":4949297992}],"flavor":"ARM_THREAD_STATE64","lr":{"value":6592504436},"cpsr":{"value":2684358656},"fp":{"value":6131068096},"sp":{"value":6131068048},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593681384},"far":{"value":0}}},{"id":1650293,"frames":[{"imageOffset":18408,"symbol":"__semwait_signal","symbolLocation":8,"imageIndex":41},{"imageOffset":53876,"symbol":"nanosleep","symbolLocation":220,"imageIndex":43},{"imageOffset":2997824,"symbol":"sleep_ms(unsigned int)","symbolLocation":64,"imageIndex":36},{"imageOffset":1995496,"symbol":"PdfRenderer::run_search(int)","symbolLocation":148,"imageIndex":36},{"imageOffset":2006052,"imageIndex":36},{"imageOffset":28564,"symbol":"_pthread_start","symbolLocation":136,"imageIndex":40},{"imageOffset":7476,"symbol":"thread_start","symbolLocation":8,"imageIndex":40}],"threadState":{"x":[{"value":4},{"value":0},{"value":1},{"value":1},{"value":0},{"value":100000000},{"value":0},{"value":0},{"value":8335707576,"symbolLocation":0,"symbol":"clock_sem"},{"value":3},{"value":17},{"value":2},{"value":0},{"value":0},{"value":0},{"value":0},{"value":334},{"value":8507026528},{"value":0},{"value":0},{"value":6131641696},{"value":4697718784},{"value":0},{"value":0},{"value":4949298112},{"value":4949297928},{"value":0},{"value":0},{"value":0}],"flavor":"ARM_THREAD_STATE64","lr":{"value":6592504436},"cpsr":{"value":2684358656},"fp":{"value":6131641680},"sp":{"value":6131641632},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593681384},"far":{"value":0}}},{"id":1650294,"frames":[{"imageOffset":7456,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":40}],"threadState":{"x":[{"value":6132215808},{"value":34563},{"value":6131679232},{"value":0},{"value":409604},{"value":18446744073709551615},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":0}],"flavor":"ARM_THREAD_STATE64","lr":{"value":0},"cpsr":{"value":4096},"fp":{"value":0},"sp":{"value":6132215808},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6593912096},"far":{"value":0}}},{"triggered":true,"id":1650295,"threadState":{"x":[{"value":0},{"value":0},{"value":36615},{"value":36615},{"value":0},{"value":0},{"value":48},{"value":0},{"value":1902103189667119336},{"value":1902103189667119336},{"value":6592944400,"symbolLocation":0,"symbol":"_ffpp_strtoencf64_l.doublePowerOf10"},{"value":4294967248},{"value":65537},{"value":0},{"value":1650295},{"value":0},{"value":6594119936,"symbolLocation":0,"symbol":"_platform_strlen"},{"value":8507029384},{"value":0},{"value":5763638368},{"value":4967495144},{"value":4967495088},{"value":0},{"value":5763638368},{"value":4340623376},{"value":0},{"value":47},{"value":0},{"value":9223372036854775791}],"flavor":"ARM_THREAD_STATE64","lr":{"value":4340623536},"cpsr":{"value":1610616832},"fp":{"value":6132788032},"sp":{"value":6132787728},"esr":{"value":2449473542,"description":"(Data Abort) byte read Translation fault"},"pc":{"value":6594119940,"matchesCrashFrame":1},"far":{"value":0}},"frames":[{"imageOffset":6404,"symbol":"_platform_strlen","symbolLocation":4,"imageIndex":37},{"imageOffset":239792,"imageIndex":36},{"imageOffset":2185332,"symbol":"sqlite3_exec","symbolLocation":404,"imageIndex":36},{"imageOffset":239200,"symbol":"DatabaseManager::select_mark(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::vector<Mark, std::__1::allocator<Mark>>&)","symbolLocation":260,"imageIndex":36},{"imageOffset":334692,"symbol":"Document::load_document_metadata_from_db()","symbolLocation":504,"imageIndex":36},{"imageOffset":368964,"imageIndex":36},{"imageOffset":512904,"imageIndex":36},{"imageOffset":28564,"symbol":"_pthread_start","symbolLocation":136,"imageIndex":40},{"imageOffset":7476,"symbol":"thread_start","symbolLocation":8,"imageIndex":40}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4622630912,
    "size" : 32768,
    "uuid" : "bab7ee3d-ce5e-37d6-8d6e-f552aea36391",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqmacjp2.dylib",
    "name" : "libqmacjp2.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4416897024,
    "size" : 32768,
    "uuid" : "f42378cf-c9c9-36b0-9c4c-5866dc40ff86",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqtga.dylib",
    "name" : "libqtga.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4416815104,
    "size" : 32768,
    "uuid" : "06500557-4537-3752-875b-707c9ad93bec",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqicns.dylib",
    "name" : "libqicns.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4443717632,
    "size" : 32768,
    "uuid" : "412c12a2-2787-3a4f-a5bd-3f787b904dbe",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqpdf.dylib",
    "name" : "libqpdf.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4671406080,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtPdf",
    "size" : 8175616,
    "uuid" : "d0be1ce6-5efc-37a8-90fb-ea9a96f28c44",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtPdf.framework\/Versions\/A\/QtPdf",
    "name" : "QtPdf",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4423991296,
    "size" : 32768,
    "uuid" : "b3999c49-94d9-387d-8813-653d62c9bf4b",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqsvg.dylib",
    "name" : "libqsvg.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4443275264,
    "size" : 393216,
    "uuid" : "237646a6-00b4-3be3-9fc3-9844626f6fdc",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqtiff.dylib",
    "name" : "libqtiff.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4423499776,
    "size" : 425984,
    "uuid" : "f3415f27-b091-3b92-8816-8ae9987fecec",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqjpeg.dylib",
    "name" : "libqjpeg.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4417241088,
    "size" : 32768,
    "uuid" : "554c616b-74bb-3aad-9311-4bfdbb6152b9",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqmacheif.dylib",
    "name" : "libqmacheif.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4417159168,
    "size" : 32768,
    "uuid" : "d6b39361-793f-3a6b-bb70-7cdb7bc6c2dc",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqico.dylib",
    "name" : "libqico.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4422991872,
    "size" : 442368,
    "uuid" : "afb09f29-c1c6-3f4a-b548-4ff410a78bed",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqwebp.dylib",
    "name" : "libqwebp.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4417077248,
    "size" : 32768,
    "uuid" : "0bb6602f-bb41-3a7a-a321-0a58c563a796",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqwbmp.dylib",
    "name" : "libqwbmp.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4416995328,
    "size" : 32768,
    "uuid" : "7016d1af-7056-3787-8937-1af22fe1c883",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/imageformats\/libqgif.dylib",
    "name" : "libqgif.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 4416651264,
    "size" : 49152,
    "uuid" : "e8a1b184-0349-3c61-a119-6543eb038317",
    "path" : "\/usr\/lib\/libobjc-trampolines.dylib",
    "name" : "libobjc-trampolines.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4416552960,
    "size" : 49152,
    "uuid" : "de7edee1-ba24-3a3d-8b1d-492180cee2ad",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/iconengines\/libqsvgicon.dylib",
    "name" : "libqsvgicon.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4410425344,
    "size" : 147456,
    "uuid" : "8f9cbcc6-87f7-3528-a266-a333677509fe",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/styles\/libqmacstyle.dylib",
    "name" : "libqmacstyle.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 4409786368,
    "CFBundleShortVersionString" : "1.0",
    "CFBundleIdentifier" : "com.apple.AppleMetalOpenGLRenderer",
    "size" : 442368,
    "uuid" : "83d94df2-aca8-3695-a3a1-bf8ca16657ac",
    "path" : "\/System\/Library\/Extensions\/AppleMetalOpenGLRenderer.bundle\/Contents\/MacOS\/AppleMetalOpenGLRenderer",
    "name" : "AppleMetalOpenGLRenderer",
    "CFBundleVersion" : "1"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4388388864,
    "size" : 655360,
    "uuid" : "24e1e52e-027b-3f8f-b02a-288279fcd6d8",
    "path" : "\/Applications\/sioyek.app\/Contents\/PlugIns\/platforms\/libqcocoa.dylib",
    "name" : "libqcocoa.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 4400611328,
    "CFBundleShortVersionString" : "3.0",
    "CFBundleIdentifier" : "com.apple.security.csparser",
    "size" : 131072,
    "uuid" : "d5ba6df5-4b0f-3b49-b7e3-923347e0a7d1",
    "path" : "\/System\/Library\/Frameworks\/Security.framework\/Versions\/A\/PlugIns\/csparser.bundle\/Contents\/MacOS\/csparser",
    "name" : "csparser",
    "CFBundleVersion" : "61123.100.169"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4390305792,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtQuickWidgets",
    "size" : 81920,
    "uuid" : "8775d2c0-7eaf-364a-96eb-7a42e61261d9",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtQuickWidgets.framework\/Versions\/A\/QtQuickWidgets",
    "name" : "QtQuickWidgets",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4402053120,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtQuick",
    "size" : 4063232,
    "uuid" : "50651b87-23e4-348b-93e1-a9011aa5a221",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtQuick.framework\/Versions\/A\/QtQuick",
    "name" : "QtQuick",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4390174720,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtOpenGLWidgets",
    "size" : 49152,
    "uuid" : "17df069b-7d3c-3473-ba95-76fcb0533ca6",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtOpenGLWidgets.framework\/Versions\/A\/QtOpenGLWidgets",
    "name" : "QtOpenGLWidgets",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4391714816,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtOpenGL",
    "size" : 409600,
    "uuid" : "71f60eb4-5bda-34ce-ae28-6150eed431fa",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtOpenGL.framework\/Versions\/A\/QtOpenGL",
    "name" : "QtOpenGL",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4392321024,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtWidgets",
    "size" : 4620288,
    "uuid" : "45cafaab-5d55-3c78-8832-7f3ef039a017",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtWidgets.framework\/Versions\/A\/QtWidgets",
    "name" : "QtWidgets",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4398399488,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.Qt3DInput",
    "size" : 327680,
    "uuid" : "235765b7-3f1b-3e91-bf20-9a55ef17dbcc",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/Qt3DInput.framework\/Versions\/A\/Qt3DInput",
    "name" : "Qt3DInput",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4398858240,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.Qt3DCore",
    "size" : 327680,
    "uuid" : "632041ad-c8d6-36f7-af3b-24212f3ea420",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/Qt3DCore.framework\/Versions\/A\/Qt3DCore",
    "name" : "Qt3DCore",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4399398912,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtSvg",
    "size" : 294912,
    "uuid" : "8ab904ac-9e21-34f4-91a2-29c710d87496",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtSvg.framework\/Versions\/A\/QtSvg",
    "name" : "QtSvg",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4389928960,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtTextToSpeech",
    "size" : 81920,
    "uuid" : "4263f052-f0d3-3ac7-8448-c644443273a2",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtTextToSpeech.framework\/Versions\/A\/QtTextToSpeech",
    "name" : "QtTextToSpeech",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4390486016,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtMultimedia",
    "size" : 606208,
    "uuid" : "4eaae195-9489-39c6-92ce-b578cf11fc7d",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtMultimedia.framework\/Versions\/A\/QtMultimedia",
    "name" : "QtMultimedia",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4424105984,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtGui",
    "size" : 6635520,
    "uuid" : "6319e5b4-cc0b-3b41-a5fe-d6de1f323bc9",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtGui.framework\/Versions\/A\/QtGui",
    "name" : "QtGui",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4401315840,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtQmlModels",
    "size" : 540672,
    "uuid" : "6e906706-64f8-3786-bc9e-e6520d871fd6",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtQmlModels.framework\/Versions\/A\/QtQmlModels",
    "name" : "QtQmlModels",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4417372160,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtQml",
    "size" : 3784704,
    "uuid" : "b8e1bbf6-cbe7-3687-9149-8e1bb9f9039d",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtQml.framework\/Versions\/A\/QtQml",
    "name" : "QtQml",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4410933248,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtNetwork",
    "size" : 1196032,
    "uuid" : "441c48bf-0071-3eb6-b856-5e7ba17fb8e1",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtNetwork.framework\/Versions\/A\/QtNetwork",
    "name" : "QtNetwork",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4390076416,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtConcurrent",
    "size" : 16384,
    "uuid" : "307c0fe6-1274-34a5-81ea-76221152515c",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtConcurrent.framework\/Versions\/A\/QtConcurrent",
    "name" : "QtConcurrent",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4444012544,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtCore",
    "size" : 4898816,
    "uuid" : "36d433b3-6671-3d70-b539-a559950fbb64",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtCore.framework\/Versions\/A\/QtCore",
    "name" : "QtCore",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4407656448,
    "CFBundleShortVersionString" : "6.7",
    "CFBundleIdentifier" : "org.qt-project.QtDBus",
    "size" : 524288,
    "uuid" : "68bb128c-d7f2-3fbe-b35e-b3ea9efacf90",
    "path" : "\/Applications\/sioyek.app\/Contents\/Frameworks\/QtDBus.framework\/Versions\/A\/QtDBus",
    "name" : "QtDBus",
    "CFBundleVersion" : "6.7.2"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4340383744,
    "CFBundleShortVersionString" : "2.0",
    "CFBundleIdentifier" : "info.sioyek.sioyek",
    "size" : 43237376,
    "uuid" : "7068a90b-3317-3d66-b28e-1c069ed0b77e",
    "path" : "\/Applications\/sioyek.app\/Contents\/MacOS\/sioyek",
    "name" : "sioyek"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6594113536,
    "size" : 32744,
    "uuid" : "031f3e8c-5227-3138-a444-68174c1c28cf",
    "path" : "\/usr\/lib\/system\/libsystem_platform.dylib",
    "name" : "libsystem_platform.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6590201856,
    "size" : 559580,
    "uuid" : "8e1e5ee2-f89a-33a7-bb0a-74bdc06b7828",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  },
  {
    "size" : 0,
    "source" : "A",
    "base" : 0,
    "uuid" : "00000000-0000-0000-0000-000000000000"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6593904640,
    "size" : 53248,
    "uuid" : "45239f06-cc53-36d0-9933-7776ac7ea2fa",
    "path" : "\/usr\/lib\/system\/libsystem_pthread.dylib",
    "name" : "libsystem_pthread.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6593662976,
    "size" : 241660,
    "uuid" : "1889ce0a-52e7-3122-8907-81af920ac472",
    "path" : "\/usr\/lib\/system\/libsystem_kernel.dylib",
    "name" : "libsystem_kernel.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6594330624,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.CoreFoundation",
    "size" : 5083136,
    "uuid" : "33908a83-098f-3437-973e-fb848c4f39df",
    "path" : "\/System\/Library\/Frameworks\/CoreFoundation.framework\/Versions\/A\/CoreFoundation",
    "name" : "CoreFoundation",
    "CFBundleVersion" : "2420"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6592450560,
    "size" : 520180,
    "uuid" : "49477e07-e77b-332f-b98d-79ca210a866d",
    "path" : "\/usr\/lib\/system\/libsystem_c.dylib",
    "name" : "libsystem_c.dylib"
  }
],
  "sharedCache" : {
  "base" : 6589448192,
  "size" : 4189880320,
  "uuid" : "664c17e3-f7ed-3226-8594-589aa9bf8fd2"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=931.0M resident=0K(0%) swapped_out_or_unallocated=931.0M(100%)\nWritable regions: Total=1.7G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.7G(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nActivity Tracing                   256K        1 \nColorSync                          608K       28 \nCoreAnimation                       16K        1 \nCoreGraphics                        16K        1 \nFoundation                          16K        1 \nKernel Alloc Once                   32K        1 \nMALLOC                             1.7G       42 \nMALLOC guard page                  192K       12 \nOpenGL GLSL                        256K        3 \nSTACK GUARD                       56.2M       12 \nStack                             13.8M       12 \nVM_ALLOCATE                        224K       14 \n__AUTH                             974K      215 \n__AUTH_CONST                      16.1M      382 \n__CTF                               824        1 \n__DATA                            5086K      407 \n__DATA_CONST                      20.4M      420 \n__DATA_DIRTY                       931K      117 \n__FONT_DATA                        2352        1 \n__GLSLBUILTINS                    5174K        1 \n__LINKEDIT                       533.4M       38 \n__OBJC_RO                         71.7M        2 \n__OBJC_RW                         2195K        1 \n__TEXT                           397.6M      434 \ndyld private memory                272K        2 \nmapped file                      182.6M       25 \nshared memory                     1360K       14 \n===========                     =======  ======= \nTOTAL                              3.0G     2188 \n",
  "legacyInfo" : {
  "threadTriggered" : {

  }
},
  "logWritingSignature" : "700d067efb4d18ab488e85563fd10b4e65a37d00",
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "639124e81d92412bfb4880b3",
      "factorPackIds" : {

      },
      "deploymentId" : 240000012
    },
    {
      "rolloutId" : "645eb1d0417dab722a215927",
      "factorPackIds" : {

      },
      "deploymentId" : 240000005
    }
  ],
  "experiments" : [

  ]
}
}

Model: Mac15,6, BootROM 10151.101.3, proc 12:6:6 processors, 36 GB, SMC 
Graphics: Apple M3 Pro, Apple M3 Pro, Built-In
Display: Color LCD, 3024 x 1964 Retina, Main, MirrorOff, Online
Memory Module: LPDDR5, Micron
AirPort: spairport_wireless_card_type_wifi (0x14E4, 0x4388), wl0: Jan 13 2024 06:19:30 version 23.30.42.0.41.51.132 FWID 01-5ba6bbe8
AirPort: 
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB31Bus
USB Device: USB31Bus
USB Device: USB31Bus
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
```

</details>

**Comment by ahrm** [comment]
Does the error happen on the preview build here: https://github.com/ahrm/sioyek/releases/tag/sioyek3-alpha0?

**Comment by SichangHe** [comment]
> Is it with fresh database files (`local.db` or `shared.db`)? Or are you using the old database files?

Those are old database files from my previous laptop. I moved both the database files and Sioyek to my new laptop, then noticed the hang issue, then downloaded the new Sioyek and discovered the SegFault.

I can try removing the database files if you want, but I do not remember where they are.

**Comment by ahrm** [comment]
I think they should be inside sioyek.app (sioyek should print them when running it from the command line). Is it possible for you to build sioyek from source?

**Comment by SichangHe** [comment]
No, the database files are instead in `~/Library/Application Support/sioyek/`, as running `sioyek` shows.

I removed the old `local.db` and now the list of previously opened files opens instantly. `local.db` was 380K; was that why it was slow?

As of building from source, I vaguely remember trying it several months back and failing.

**Comment by ahrm** [comment]
It is possible the old database files are causing compatibility issues. Does the new sioyek version crash when you move the database files?

**Comment by SichangHe** [comment]
Okay, if I remove `local.db`, the sioyek3-alpha0 build opens and seems to run normally. Is there any way to migrate the database file?

However, now I recall why I switched back to the older Release v0.31.433 on my previous laptop: copying selection in the newer versions do not work.

**Comment by ahrm** [comment]
I have made a script here that migrates the database: https://github.com/ahrm/sioyek/discussions/1156#discussioncomment-10614340. Of course this is only for the preview version, in the final version we do this migration automatically, but for now you have to run in manually.

**Comment by SichangHe** [comment]
Thanks for the pointer. I migrated `shared.db` and put `local.db` back. Now it does not SegFault on start, but still hangs for a while every time I open the list of previously-viewed files. Is `local.db` to be migrated as well?

**Comment by ahrm** [comment]
Hmmm, if you don't have private data in your files could you email me your database files so I can see if there is something wrong? Alternatively, if you have private data, you could open the database files yourself using a sqlite database browser and report if there is anything weird. For example the number of rows in each table could be relevant.

**Comment by SichangHe** [comment]
I guess it is Qt taking forever to render the scrolldown menu, since it also freezes my tiling window manager Yabai (macOS built-in display server is alpha quality).

Yes, I can send you the database files if that helps troubleshooting.

**Comment by SichangHe** [comment]
[sichanghe_sioyek_dbs.zip](https://github.com/user-attachments/files/18219300/sichanghe_sioyek_dbs.zip)

Please just delete this comment after downloading my files, @ahrm.

**Comment by ahrm** [comment]
I was able to reproduce the long load times when opening the previous documents list, the issue is indeed qt taking a long time to render the huge list. In the future versions of sioyek we will use a custom painting function which should speed up large lists. For now, do you have the `multiline_menus` config enabled? In my experience disabling that config significantly speeds up the menu.

**Comment by SichangHe** [comment]
Thanks for checking it out and the tip. Setting `multiline_menus 0` does not improve performance for me. I guess I will just clean up my history.

I will open a separate issue for the inability to copy.

**Comment by SichangHe** [comment]
Okay, sorry for the stupidity regarding copying. I just needed to press <kbd>ctrl</kbd> <kbd>c</kbd> instead of <kbd>command</kbd> <kbd>c</kbd>.

**Comment by SichangHe** [comment]
> I guess I will just clean up my history.

It is still very slow. I reduced the list to around 40 and it still takes 3 seconds to open. This seems like a regression somewhere because it used to be fast.

**Comment by ahrm** [comment]
That's strange, with 40 files it should be instantaneous. Unfortunately I can not reproduce this issue, for me it is very fast even with 500 files. If you could build sioyek maybe you could do a git bisect and find out the commit that causes the issue (building the development branch is similar to the main branch, only we use Qt 6.7+ in the development branch).

**Comment by SichangHe** [comment]
Looks like it is a Qt problem on this new Mac. I built development and main branches and both are slow. The official 2.0.0 release from 2022 is also slow.

Is there an easy way to profile what code is using all this time? Or am I better off just putting up with it?

**Comment by SichangHe** [comment]
I installed the newer Qt 6.8.1 and built Sioyek with it, but the problem persists, so it is still not resolved by Qt since macOS 14 came out.

**Comment by ahrm** [comment]
> Is there an easy way to profile what code is using all this time? Or am I better off just putting up with it?

You can profile the code using a profiler. Personally I use visual studio on windows, but I am sure there are alternatives like valgrind on mac. Alternatively you can add timing logs to `handle_open_prev_doc` here: https://github.com/ahrm/sioyek/blob/bab683c46f92421427f505377ff853d2a1905200/pdf_viewer/main_widget.cpp#L5939.

Is it possible that your database files are on a slower drive now? Like a shared network folder or an slow external drive?

**Comment by SichangHe** [comment]
<img width="2160" alt="image" src="https://github.com/user-attachments/assets/a8dd6945-75e8-4edc-9765-4c4bafd92cf4" />

All the time is spent checking if each file exists at https://github.com/ahrm/sioyek/blob/159304abef7eb0c77bf7892a305d60bea19a997b/pdf_viewer/checksum.cpp#L66

**Comment by SichangHe** [comment]
> Is it possible that your database files are on a slower drive now?

I've made sure all the files appearing in `local.db` is on my MacBook and not in a cloud drive, so that does not seem to be it.

**Comment by ahrm** [comment]
So is the performance issue fixed now? If not, I could add a config option to skip checking for the existence of files, but that would mean that we could show non-existent files in the files menu.

**Comment by SichangHe** [comment]
Yes, it is instant now.

As for the fix, ideally I would do an async check and remove files from the list if they do not exist, but this is C++, so that may be unnecessarily difficult.
I would suggest a timeout on each existence check instead.
Also, it may be sensible to delete entries of nonexistent files in the database to make it a little more sustainable.

Thanks!

**Comment by ahrm** [comment]
The problem is the non-existent files might be temporarily non-existent. For  example, in your case you might later re-open your google drive, or an external drive might be temporarily disconnected from a computer.
